libminifi/include/core/PropertyValidation.h

```diff
@@ -259,6 +259,22 @@ class UnsignedLongValidator : public PropertyValidator {
   }
 };
 
+class NonBlankValidator : public PropertyValidator {
+ public:
+  explicit NonBlankValidator(const std::string& name)
+      : PropertyValidator(name) {
+  }
+  ~NonBlankValidator() override = default;
+
+  ValidationResult validate(const std::string& subject, const std::shared_ptr<minifi::state::response::Value>& input) const final {
+    return validate(subject, input->getStringValue());
+  }
+
+  ValidationResult validate(const std::string& subject, const std::string& input) const final {
+    return ValidationResult::Builder::createBuilder().withSubject(subject).withInput(input).isValid(utils::StringUtils::trimLeft(input).size()).build();
```

Comment:
I would prefer and explicit `.size() != 0`

Reply:
Should not be a factor in readibility.